### PR TITLE
:robot: Update gradle-build-deploy.yml

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
#6 

### What
Rename workflow
Add aws action

### Why
Will enable auto-deploy 

### Testing
Will do

### Additional Notes
Crossed fingers.

### CODE REVIEW

File: .github/workflows/gradle-build-deploy.yml [hunk 1]
```diff
@@ -29,3 +29,20 @@ jobs:
       with:
         name: website
         path: build/site
+        
+
+  deploy:
+    needs:
+      - release
+    runs-on: ubuntu-latest
+    steps:
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:38]: Definitely needs an `if`. You don't want this to run for every pull request, rather for pushes to main. [duckbot](https://github.com/Chippers255/duckbot/blob/main/.github/workflows/python-package.yml#L54)'s is a decent template.

File: .github/workflows/gradle-build-deploy.yml [hunk 2]
```diff
@@ -29,3 +29,20 @@ jobs:
...
+      - name: Configure AWS credentials from Production account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ca-central-1
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:44]: region should probably be a secret as well

File: .github/workflows/gradle-build-deploy.yml [hunk 3]
```diff
@@ -29,3 +29,20 @@ jobs:
...
+
+      - name: Copy files to the website with the AWS CLI
+        run: |
+          aws s3 sync ./build/site/* s3://modsurski.com/csci162
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:48]: These files don't exist in this job. The `release` job uploads them as an artifact though, you can [download](https://github.com/actions/download-artifact) with a sister action.

File: .github/workflows/gradle-build-deploy.yml [hunk 4]
```diff
@@ -29,3 +29,20 @@ jobs:
...
+  deploy:
+    needs:
+      - release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS credentials from Production account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:43]: These creds, how much does the user/role have access to? May want to make a new role with only access to the specific prefix in the s3 bucket to isolate breaches if they happen.

File: .github/workflows/gradle-build-deploy.yml (hunk 1 above)
Comment (twentylemon) [reply .github/workflows/gradle-build-deploy.yml]: You'll also want `concurrency` here. It'll prevent two deploy steps running at the same time, so you never publish an out of date version.  
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency

File: .github/workflows/gradle-build-deploy.yml [hunk 5]
```diff
@@ -30,20 +30,29 @@ jobs:
...
+    concurrency: deploying
+    steps: 
+    - uses: actions/checkout@v2
+    - uses: actions/download-artifact@v2
+      with:
+        name: website
+    - name: Unzip website
+      run: unzip website.zip
+    - name: Remove website.zip
+      run: rm website.zip
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:47]: If it doesn't interfere with the rest of the job, I'd say leave the zip file there. Actions runners are ephemeral anyways, they'll clean it up.

File: .github/workflows/gradle-build-deploy.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      run: unzip website.zip -d ./site
+        
+    - name: Configure AWS credentials from Production account
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ secrets.AWS_REGION }}
+
+    - name: Copy files to the website with the AWS CLI
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:56]: Minor, step names, going to gnit pick.
* title case stuff, similar to the `release` job (_Set up Java_ not _Set up java_)
* folder -> directory
* remove _from Production account_ -- you don't have any other stage
* remove _using aws cli_ -- names should describe what, not how

**Review by twentylemon — COMMENTED** [review]
Looks fine overall. Would like to go over the iam role permissions offline (meaning, discord) before approval.

File: .github/workflows/gradle-build-deploy.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        
+  deploy:
+    needs:
+    - release
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.repository == 'jameshughes89/cs102'
+    concurrency: deploying
+    steps: 
+    - uses: actions/checkout@v2
+    - uses: actions/download-artifact@v2
```
Comment (twentylemon) [inline .github/workflows/gradle-build-deploy.yml:41]: Remove checkout, the repo code isn't actually used. Also, prefer to name the download step.

**Review by twentylemon — APPROVED** [review]
LGTM

**Comment by jameshughes89** [comment]
> LGTM

:sun_with_face:

**Review by twentylemon — APPROVED** [review]

**Pull request merged**